### ISSUE TITLE
Switching between "Program" mode, and "Override" mode

### ISSUE THREAD

**Issue by MachRat** [andrewleech/ha_magiqtouch#17]
Hi Andrew,

I just installed this today, and it works great. 
Thank you for the effort in creating and sharing this integration.

Using the Seely iphone app I can flip between "Program" mode (heating is controlled by the schedule I setup in the wall unit), and "Override" mode (heating is then controlled by my Home Assistant setup).

Quick question as I may be misinterpreting something. 
Is it possible to switch between "Program" mode and "Override" mode from within Home Assistant?

Thanks again for your innovation and efforts

**Comment by andrewleech** [comment]
I've never seen / used an override mode.... or a schedule set from the wall panel, so I guess no there isn't any feature in the integration to control this setting.

**Comment by MachRat** [comment]
Yeah, maybe has something to do with `ProgramModeOverridden` in class RemoteStatus, and/or override in class 'RemoteAccessRequest'.  My wild guess would be that the iphone app toggles this to off to put the wall unit in "program mode" to use its built in scheduling capability. In program mode the wall unit controls the temp (home assistant becomes read only for temperature). When toggled to override mode the unit ignores the program in the wall unit (and then the API / home assistant can read AND also change the temperature). At least this is how the mobile app appears to behave, and I assume it uses the same API so must be something in there somewhere...

But in any case, HA is working to adjust the heater temp if i toggle it to override mode in the app so all good. 
Thanks again

**Comment by Patrick384600** [comment]
Same issue here — HA commands are overridden by the wall controller schedule. The ProgramModeOverridden field in the API appears to be the key. Would be great to expose this as a switch entity in HA so we can toggle Override mode without needing the app.

**Comment by BradJMelb** [comment]
I've just started playing with HA for the first time after wanting to try ha_magiqtouch for a long time. I'm finding the same thing when the wall thermostat is in program mode the status in HA is correct but I cannot control/override the settings. I can switch between heating/cooling/fan and off but temperature changes don't do anything.
We've been using the built-in scheduling since installing the unit. We could instead run in manual mode and set the schedule in HA but it would be nice if the controls can be exposed to HA. Could you do an override feature or support switching the mode from Program to Manual (or even both)?
Great work so far though, the manufactures phone app is very clunky so this is looking very positive.

Thanks.

**Comment by BradJMelb** [comment]
I’ve completed some fairly detailed testing of Program mode, per-zone Override, Manual mode and Fan Only against the cloud protocol. I have a working local prototype for some of this, although I’m not a developer and haven’t prepared a pull request.

Environment:

- Upstream base: `ha_magiqtouch` v5.2
- Home Assistant: 2026.9.0

Confirmed Program Override behaviour:

- The cloud state contains `programMode`, `ProgramModeOverridden` and `ProgramPeriodActive` on both the heater and cooler records for each zone.
- A reliable Override command needs to enable Override on both the heater and cooler copies of the zone before setting the requested temperature on both records.
- Sending the target only to the heater record is rejected and the scheduled target returns.
- Turning Override off restores the current scheduled zone state and target.
- The wall controller remembers the most recent override temperature. Re-enabling Override restores that temperature.
- Override temperatures changed at the wall panel can be learned from the cloud state.
- Inactive scheduled zones report 27°C as a placeholder. Remembering the last meaningful target produces a more useful HA display.

Our local prototype successfully provides one Program Override switch per zone. Changing a zone target while Program is active automatically enables Override, and cancelling it restores the schedule. The remembered override temperature also survives an HA restart.

Manual/Program findings:

- Changing Manual/Program at the wall panel is reported through `programMode` in the cloud status.
- We captured the complete state transition for both heater and cooler records, including zone on/off states, targets, fan speed, override flags and running state.
- We attempted to send a Manual payload matching the panel-generated state field-for-field.
- The controller received the request and incremented `touchCount`, but returned with Program mode unchanged.
- This suggests `programMode` is exposed as status but is read-only, or requires a different command/endpoint from the normal full-state update.

We therefore abandoned the writable Manual/Program selector. A read-only mode sensor would still be possible.

Fan Only finding:

- During `HEATER_FAN` or `COOLER_FAN`, the API reports the correct numeric `fan_speed` but leaves `control_mode` as `TEMP`.
- The existing integration checks `control_mode` first, so HA reports `fan_mode: Temperature` instead of the numeric speed.
- Special-casing those two running modes made HA correctly track fan speeds 1–10.
- However, leaving Fan Only for Heat/Cool puts the controller into Manual mode. The official MagIQtouch app behaves the same way, so I have not pursued fan-speed control further.

Previous/next scheduled-period controls have not yet been captured or implemented.

I’m happy to provide a cleaned-up diff of the working Program Override changes, the fan-speed reporting change, or sanitised excerpts from the test logs if that would help.

**Comment by Patrick384600** [comment]
Thanks @BradJMelb for the detailed work here — really appreciate you picking this up, especially since it follows on from what I flagged back in June (Patrick384600, same person).

We're going to run the same tests against our own unit (enable debug logging, toggle Program/Override at the wall panel, capture the raw cloud payloads) to independently confirm your findings before anything gets merged — will report back once we have real data from a second household. Thanks for the offer of the diff too, we'll take you up on that once we've done our own pass.

**Comment by BradJMelb** [comment]
Update from Seeley Technical Support after I asked them about functions of their app:

Seeley has confirmed that the official MagIQtouch app intentionally provides only basic remote functions and does not support:

- Switching remotely between Program and Manual.
- Returning directly to Program after Fan Only has exited Program mode.
- Moving to the previous or next scheduled period.

They have passed the requests to their development team, but advised that there are currently no plans to change the app.

I guess this doesn’t prove that no undocumented cloud command exists, but it is consistent with our testing: programMode changes are reported when made at the wall controller, while equivalent writes through the integration’s existing command path are received but ignored.

Per-zone Program Override therefore appears to be the most useful and realistically supportable enhancement at present.

Note, if it wasn't obvious, I'm not a developer, I've been working this through with ChatGPT so that's why my notes refer to 'our' testing. I just wanted to put that out there.
